components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -99,24 +99,30 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
      * This performs network requests, and should not be used on the main thread.
      *
      * @param scopes List of OAuth scopes for which the client wants access
+     * @param entrypoint to be used for data collection
      * @return String that resolves to the flow URL when complete
      */
-    fun beginOAuthFlow(scopes: Array<String>): String {
+    fun beginOAuthFlow(scopes: Array<String>, entrypoint: String): String {
         val scope = scopes.joinToString(" ")
         return rustCallWithLock { e ->
-            LibFxAFFI.INSTANCE.fxa_begin_oauth_flow(this.handle.get(), scope, e)
+            LibFxAFFI.INSTANCE.fxa_begin_oauth_flow(this.handle.get(), scope, entrypoint, e)
         }.getAndConsumeRustString()
     }
 
     /**
      * Begins the pairing flow.
      *
      * This performs network requests, and should not be used on the main thread.
+     *
+     * @param pairingUrl the url to initilaize the paring flow with
+     * @param scopes List of OAuth scopes for which the client wants access
+     * @param entrypoint to be used for data collection
```

Comment:
- Let's make this an optional parameter
- change `data collection` to `metrics`